angularjs/angular-animate.d.ts

```diff
@@ -14,27 +14,68 @@ declare module "angular-animate" {
  * ngAnimate module (angular-animate.js)
  */
 declare module angular.animate {
-    interface IAnimateFactory extends Function {
-        enter?: (element: ng.IAugmentedJQuery, doneFn: Function) => IAnimateCssRunner|void;
-        leave?: (element: ng.IAugmentedJQuery, doneFn: Function) => IAnimateCssRunner|void;
-        addClass?: (element: ng.IAugmentedJQuery, className: string, doneFn: Function) => IAnimateCssRunner|void;
-        removeClass?: (element: ng.IAugmentedJQuery, className: string, doneFn: Function) => IAnimateCssRunner|void;
-        setClass?: (element: ng.IAugmentedJQuery, className: string, doneFn: Function) => IAnimateCssRunner|void;
+    interface IAnimateFactory {
+        (...args: any[]): IAnimateCallbackObject;
     }
 
+    interface IAnimateCallbackObject {
+        eventFn?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        setClass?: (element: IAugmentedJQuery, addedClasses: string, removedClasses: string, doneFunction: Function, options: IAnimationOptions) => any;
+        addClass?: (element: IAugmentedJQuery, addedClasses: string, doneFunction: Function, options: IAnimationOptions) => any;
+        removeClass?: (element: IAugmentedJQuery, removedClasses: string, doneFunction: Function, options: IAnimationOptions) => any;
+        enter?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        leave?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        move?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        animate?: (element: IAugmentedJQuery, fromStyles: string, toStyles: string, doneFunction: Function, options: IAnimationOptions) => any;
```

Comment:
Yes, `executeAnimationFn` returns a function. But it has nothing (well, almost…) to do with `IAnimateCallbackObject` interface :)
`executeAnimationFn` takes `fn` as the first argument and `fn` here is one of the functions in `IAnimateCallbackObject` interface. `fn` could be `setClass`, `addClass`, `removeClass` and so on. In other words the return type of functions inside the `IAnimateCallbackObject` interface (currently annotated as `any`) is the same type as variable `value` has [on line 170](https://github.com/angular/angular.js/blob/master/src/ngAnimate/animateJs.js#L170).

Therefore reading the code below line 170 we can deduce the type of variable `value` (this type could replace `any` in the `IAnimateCallbackObject` interface):
* void
* an instance of $$AnimateRunner
* a function
* an object with `start` method which might return:
  * void
  * an instance of $$AnimateRunner
  * function

The problem is: **This is not documented anywhere** and thus it's prone to breaking changes, moreover I doubt this information is anyhow useful – nobody complained so far.
The safest and future-proof solution is to keep these functions return types annotated as `any`. This can be easily updated if documentation gets updated at docs.angularjs.org.